File: src/main/java/com/fifo/ticketing/domain/book/dto/BookUserDetailDto.java
package com.fifo.ticketing.domain.book.dto;

import com.fifo.ticketing.domain.book.entity.BookStatus;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@RequiredArgsConstructor
public class BookUserDetailDto {

    private final Long bookId; //bookId
    private final Long performanceId;
    private final String title;
    private final Integer totalPrice;
    private final Integer quantity;
    private final String username;
    private final String encodedFileName;
    private final BookStatus bookStatus;

    @Setter
    private String urlPrefix;

    public String getUrl() {
        return urlPrefix + encodedFileName;
    }

}

File: src/main/java/com/fifo/ticketing/domain/book/dto/BookUserDetailDto.kt
package com.fifo.ticketing.domain.book.dto

import com.fifo.ticketing.domain.book.entity.BookStatus
import lombok.Setter

class BookUserDetailDto constructor(
    val bookId: Long,
    val performanceId: Long,
    val title: String,
    val totalPrice: Int,
    val quantity: Int,
    val username: String,
    val encodedFileName: String,
    val bookStatus: BookStatus,

    ) {
    var urlPrefix: String = ""

    fun getUrl(): String = "$urlPrefix$encodedFileName"
}
